experiments/experiment.go

```diff
@@ -132,9 +144,56 @@ func (ec *experimentContext) reconcileTemplate(template v1alpha1.TemplateSpec) {
 			}
 		}
 	} else {
+		if desiredReplicaCount == 0 {
+			// Add scaleDownDelay if necessary
+			rsIsUpdated, err := ec.addScaleDownDelay(rs)
+			if err != nil {
+				ec.log.Warnf("Unable to add scaleDownDelay label on rs '%s'", rs.Name)
+			} else {
+				if rsIsUpdated {
+					ctx := context.TODO()
+					modifiedRS, err := ec.kubeclientset.AppsV1().ReplicaSets(ec.ex.Namespace).Get(ctx, rs.Name, metav1.GetOptions{})
+					if err != nil {
+						ec.log.Warnf("Unable to get rs '%s' with added scaleDownDelay", rs.Name)
+					} else {
+						rs = modifiedRS
+						ec.templateRSs[template.Name] = modifiedRS
+					}
+				}
+			}
+
+		}
```

Comment:
The addition of service creation to `reconcileTemplate` has nearly doubled the length of this method, making it quite complex.

Can we refactor this to better organize service creation into its own set of method(s)? This should make it easier to unit test as well.